### PR TITLE
Show indicator when loading slots and show toast in case of error

### PR DESCRIPTION
**Pull request by tcitworld**
Demo (before changing icon to material)

https://user-images.githubusercontent.com/2197836/173613401-972af52b-6ad0-4d68-a7fe-0257ececaae3.mp4

### CODE REVIEW

File: src/views/Appointments/Booking.vue [hunk 1]
```diff
@@ -260,6 +272,10 @@ export default {
 		flex-grow: 2;
 	}
 
+  &__slot-selection .material-design-icon.loading-icon.animation-rotate {
```
Comment (st3iny) [inline src/views/Appointments/Booking.vue:275]: Indentation is weird

**Review by st3iny — COMMENTED** [review]
Awesome!

**Comment by raimund-schluessler** [comment]
I thought about having a `LoadingIcon` component in the nextcloud-vue components to share between apps. I created a PR a while ago already, see https://github.com/nextcloud/nextcloud-vue/pull/2727. The component itself is good to go, I think. I just wanted to add some documentation to the PR and haven't had time yet.

**Comment by raimund-schluessler** [comment]
I finished the PR in the vue-components for the loading icon, see https://github.com/nextcloud/nextcloud-vue/pull/2727
Might be good to use it here.

**Comment by miaulalala** [comment]
Hm, I dislike the error toast for the slots being unable to be fetched. There is nothing gained from showing this to the user as they can't do anything about it. The neccessary info is there already with "No slots available". This will be anonoymous users a lot of the time.

**Comment by tcitworld** [comment]
> they can't do anything about it

The difference between « no slots available » and « an error has occurred » is that they can report it to whoever gave them the link. If it's just a timeout issue like in https://github.com/nextcloud/calendar/issues/4637#issuecomment-1282128225 it's possible it won't be noticed by the server admin.

**Comment by miaulalala** [comment]
> > they can't do anything about it
> 
> The difference between « no slots available » and « an error has occurred » is that they can report it to whoever gave them the link. If it's just a timeout issue like in [#4637 (comment)](https://github.com/nextcloud/calendar/issues/4637#issuecomment-1282128225) it's possible it won't be noticed by the server admin.

That's true if it's an internal user booking a slot, but for external users they won't bother. It just gives the impression that things don't work.

**Comment by joshtrichards** [comment]
> That's true if it's an internal user booking a slot, but for external users they won't bother. It just gives the impression that things don't work.

Yeah but not showing it gives the impression to the external user that there are never any slots and they can never book an appointment. :-) At least making it apparent there was an error (versus saying "no slots" after a failed fetch) means the external user has the information to decide whether to reach out via some other channel to book and/or report the issue.

File: src/views/Appointments/Booking.vue (hunk 1 above)
Comment (st3iny) [reply src/views/Appointments/Booking.vue]: Done

**Review by st3iny — APPROVED** [review]
Tested and works.

I rebased the code, fixed all conflicts and pending issues.

**Pull request merged**